core/src/test/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbeTest.java

```diff
@@ -0,0 +1,219 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.Map;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.PersonIdent;
+import org.junit.jupiter.api.Test;
+
+public class HasUnreleasedProductionChangesProbeTest extends AbstractProbeTest<HasUnreleasedProductionChangesProbe> {
+    @Override
+    HasUnreleasedProductionChangesProbe getSpy() {
+        return spy(HasUnreleasedProductionChangesProbe.class);
+    }
+
+    @Test
+    void shouldBeExecutedAfterLastCommitDateProbe() {
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final HasUnreleasedProductionChangesProbe probe = getSpy();
+
+        assertThat(probe.apply(plugin, ctx))
+            .usingRecursiveComparison()
+            .comparingOnlyFields("id", "status")
+            .isEqualTo(ProbeResult.error(HasUnreleasedProductionChangesProbe.KEY, ""));
+        verify(probe, never()).doApply(plugin, ctx);
+    }
+
+    @Test
+    void shouldCheckIfTheUnreleasedCommitsExist() throws IOException, GitAPIException {
```

Comment:
This test case fails. It throws the following error:

```
  ProbeResult[id=unreleased-production-changes, message=Last commit cannot be extracted. Please validate sub-folder if any., status=FAILURE, timestamp=2023-05-18T21:55:28.761664600+05:30[Asia/Calcutta]]

```

It is because I have not added code for paths other than `pom.xml` and `src/main/**`? 

I am not sure here.